vllm/model_executor/layers/logits_processor.py

```diff
@@ -86,11 +86,17 @@ def _apply_logits_processors(
 ) -> torch.Tensor:
     logits_row_idx = 0
     found_logits_processors = False
-    for seq_ids, sampling_params in sampling_metadata.seq_groups:
+    for i, seq_group in enumerate(sampling_metadata.seq_groups):
+        seq_ids, sampling_params = seq_group
         logits_processors = sampling_params.logits_processors
         if logits_processors:
             found_logits_processors = True
             for seq_id in seq_ids:
+                # handle prompt_logprobs by skipping rows in logits added for
+                # the prompt tokens (prompt logprobs are not processed)
+                if (i < sampling_metadata.num_prompts
+                        and sampling_params.prompt_logprobs is not None):
+                    logits_row_idx += sampling_metadata.prompt_lens[i] - 1
```

Comment:
Actually @huyiwen I think this should be moved higher, prior to the `if logits_processors:` line.

In the prompt case, `len(seq_ids)` should always be 1 (you could also add an assert like @tjohnson31415 did [here](https://github.com/vllm-project/vllm/pull/3876/files#diff-a49206d5c29b28f53c18c4c18265a89eb06f11102b2f4d6ee95123ea788a602eR122)).